packages/common/src/failure.ts

```diff
@@ -35,14 +35,49 @@ checkExtends<RetryState, temporal.api.enums.v1.RetryState>();
 
 export type WorkflowExecution = temporal.api.common.v1.IWorkflowExecution;
 
+const isLineageTrackingError = Symbol.for('__temporal_is_lineage_tracking_error__');
+
+/**
+ * Base error class used to track the class lineage of any of its subclasses.
+ *
+ * This error is never directly instantiated but is used throughout the SDK code to support an alternative form of
+ * instanceof checks that works when multiple versions of the temporalio package are present.
+ *
+ * See {@link LineageTrackingError.is} and the `is` method on each subclass.
+ */
+export class LineageTrackingError extends Error {
+  public readonly name: string = 'LineageTrackingError';
+  /**
+   * Marker to determine whether an error is an instance of LineageTrackingError.
+   */
+  protected readonly [isLineageTrackingError] = true;
+  /**
+   * Array that stores subclass names.
+   * @hidden
+   */
+  public readonly lineage = Array<string>();
+
+  constructor(message?: string) {
+    super(message ?? undefined);
+  }
+
+  /**
+   * Instanceof check that is works when multiple versions of @temporalio/common are installed.
+   */
+  static is(error: unknown): error is LineageTrackingError {
+    // Cast to any since isLineageTrackingError is intentionally left protected.
+    return error instanceof LineageTrackingError || (error instanceof Error && (error as any)[isLineageTrackingError]);
```

Comment:
I think I would skip the `error instanceof Error && ` check. At that point, the presence of the symbol is enough.